tests/test_middleware.py

```diff
@@ -259,7 +259,7 @@ def test_multi_args_callbacks(settings):
     item, _, _ = yield crawl_single_item(MultiArgsCallbackSpider, ProductHtml, settings)
     assert type(item["product"]) == ProductPage
     assert type(item["provided"]) == ProvidedWithDeferred
-    assert item["cb_arg"] == "arg!"
+    assert item["cb_arg"] is None
```

Comment:
Why is this value dropped, and not preserved?

Reply:
It's because we're not allowing parameters from `Request.cb_kwargs` to replace things in the callbacks anymore.

Do you think we should have a special behavior for `Request.cb_kwargs` to only replace the callback values when they're `None`?